libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Analytics/SFSDKAILTNPublisher.m

```diff
@@ -53,41 +50,18 @@ + (BOOL) publish:(NSArray *) events {
-    NSURL *loggingEndpointUrl = [NSURL URLWithString:[NSString stringWithFormat:@"%@/%@/%@/%@", credentials.apiUrl, kRestApiPrefix, kApiVersion, kRestApiSuffix]];
-    NSString *token = [NSString stringWithFormat:kBearer, credentials.accessToken];
-    NSString *userAgent = [SalesforceSDKManager sharedManager].userAgentString(@"");
-    __block NSError *error = nil;
-    NSHTTPURLResponse* (^makeSynchronousRequest)(NSError**) = ^NSHTTPURLResponse* (NSError** error) {
-        NSMutableURLRequest *request = [NSMutableURLRequest requestWithURL:loggingEndpointUrl
-                                                               cachePolicy:NSURLRequestReloadIgnoringCacheData
-                                                           timeoutInterval:60.0];
-        [request setHTTPMethod:@"POST"];
-        [request setValue:userAgent forHTTPHeaderField:@"User-Agent"];
-        [request setValue:@"false" forHTTPHeaderField:@"X-Chatter-Entity-Encoding"];
-        [request setValue:token forHTTPHeaderField:@"Authorization"];
-        [request setValue:@"application/json" forHTTPHeaderField:@"Content-Type"];
+    NSString *path = [NSString stringWithFormat:@"/%@/%@", kSFRestDefaultAPIVersion, kRestApiSuffix];
+    SFRestRequest *request = [SFRestRequest requestWithMethod:SFRestMethodPOST path:path queryParams:nil];
 
-        // Adds GZIP compression.
-        NSString *bodyString = [[self class] dictionaryAsJSONString:bodyDictionary];
-        NSData *bodyData = [bodyString dataUsingEncoding:NSUTF8StringEncoding];
-        NSData *postData = [bodyData gzipDeflate];
-        [request setValue:@"gzip" forHTTPHeaderField:@"Content-Encoding"];
-        [request setValue:[NSString stringWithFormat:@"%lu", (unsigned long)[postData length]] forHTTPHeaderField:@"Content-Length"];
-        [request setHTTPBody:postData];
-        NSHTTPURLResponse* response = nil;
-        [NSURLConnection sendSynchronousRequest:request returningResponse:&response error:error];
-        return response;
-    };
-    NSHTTPURLResponse *response = makeSynchronousRequest(&error);
-    NSInteger code = [response statusCode];
-    if (error) {
-        [SFLogger log:[self class] level:SFLogLevelError format:@"Upload failed %ld %@", (long)[error code], [error localizedDescription]];
-        return NO;
-    } else if (code >= 200 && code < 300) {
-        return YES;
-    }
-    return NO;
+    // Adds GZIP compression.
+    NSString *bodyString = [[self class] dictionaryAsJSONString:bodyDictionary];
+    NSData *bodyData = [bodyString dataUsingEncoding:NSUTF8StringEncoding];
+    NSData *postData = [bodyData gzipDeflate];
+    [request setCustomRequestBodyData:postData contentType:@"application/json"];
+    [request setHeaderValue:@"gzip" forHeaderName:@"Content-Encoding"];
+    [request setHeaderValue:[NSString stringWithFormat:@"%lu", (unsigned long)[postData length]] forHeaderName:@"Content-Length"];
+    [[SFRestAPI sharedInstance] send:request delegate:self];
+    return YES;
```

Comment:
Maybe the network sdk (or SFRestAPI) should handle gzip-ing.
For instance, we could add a property on SFRestRequest and have SFRestAPI look for it and do what you are doing above. Or it could move lower.